app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -63,10 +63,10 @@ def distribution_repositories(host)
         lifecycle_environment = host.content_facet.try(:lifecycle_environment)
 
         if content_view && lifecycle_environment
-          Katello::Repository.where(:distribution_version => host.os.release,
-                                    :distribution_arch => host.architecture.name,
-                                    :distribution_bootable => true
-                                    )
+          Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
```

Comment:
Should we add a unit test to ensure this logic since it is quite a change from the previous incorrect assumption?